diff --git a/Application.cmake b/Application.cmake
index 1ead314..09aed85 100644
--- a/Application.cmake
+++ b/Application.cmake
@@ -19,7 +19,11 @@ ENDIF()

 #linux
 IF(UNIX AND NOT CYGWIN)
+
     SET(QWT_BUILD_DIR "/usr/local/qwt-6.4.0-ma")
+    IF(DEFINED ENV{NIX_QWT})
+        SET(QWT_BUILD_DIR "$ENV{NIX_QWT}")
+    ENDIF()
     SET(CMAKE_SKIP_BUILD_RPATH FALSE)
     SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
     SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
